dds/DCPS/RTPS/TypeLookup.h

```diff
@@ -0,0 +1,173 @@
+#ifndef OPENDDS_RTPS_TYPE_LOOKUP_H_
+#define OPENDDS_RTPS_TYPE_LOOKUP_H_
+
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "RtpsRpcTypeSupportImpl.h"
+#include <dds/DCPS/XTypes/TypeObject.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+// As per chapter 7.6.3.3.3 of XTypes spec
+// Used in TypeLookup_Call and TypeLookup_Return
+const CORBA::ULong TypeLookup_getTypes_HashId = 25318099U;
+const CORBA::ULong TypeLookup_getDependencies_HashId = 95091505U;
+
+struct TypeLookup_getTypes_In
+{
+  TypeIdentifierSeq type_ids;
+
+  TypeLookup_getTypes_In() {}
+};
+
+struct TypeLookup_getTypes_Out
+{
+  TypeIdentifierTypeObjectPairSeq types;
+  TypeIdentifierPairSeq complete_to_minimal;
+
+  TypeLookup_getTypes_Out() {}
+};
+
+struct TypeLookup_getTypes_Result
+{
+  CORBA::ULong return_code;
+  TypeLookup_getTypes_Out result;
+
+  TypeLookup_getTypes_Result() {}
+};
+
+template <typename T>
+struct SequenceOct32 : public Sequence<T>
```

Comment:
```suggestion
struct SequenceOct32 : Sequence<T>
```
I like the use of inheritance since it does need to be a distinct type and not just a typedef.  But instead of introducing two names (SequenceOct32 and OctetSeq32), just use the OctetSeq32 name here.  That will let you remove the typedef below.